src/Adapter/Order/OrderDetailUpdater.php

```diff
@@ -0,0 +1,471 @@
+
+                $taxCalculator = $this->getTaxCalculatorByAddress($taxAddress, $orderDetail);
+                $taxesAmount = $taxCalculator->getTaxesAmount($orderDetail->unit_price_tax_excl);
+                $unitAmount = $totalAmount = 0;
+                if (!empty($taxesAmount)) {
+                    $orderDetailTaxes = [];
+                    foreach ($taxesAmount as $taxId => $amount) {
+                        switch ($roundType) {
+                            case Order::ROUND_ITEM:
+                                $unitAmount = (float) Tools::ps_round($amount, $computingPrecision);
+                                $totalAmount = $unitAmount * $orderDetail->product_quantity;
+
+                                break;
+                            case Order::ROUND_LINE:
+                                $unitAmount = $amount;
+                                $totalAmount = Tools::ps_round($unitAmount * $orderDetail->product_quantity, $computingPrecision);
+
+                                break;
+                            case Order::ROUND_TOTAL:
+                                $unitAmount = $amount;
+                                $totalAmount = $unitAmount * $orderDetail->product_quantity;
+
+                                break;
+                        }
+                        $orderDetailTaxes[] = [
+                            'id_order_detail' => $orderDetail->id,
+                            'id_tax' => $taxId,
+                            'unit_amount' => (float) $unitAmount,
+                            'total_amount' => (float) $totalAmount,
+                        ];
+                        $orderDetail->unit_price_tax_incl = (float) $orderDetail->unit_price_tax_excl + $unitAmount;
+                        $orderDetail->total_price_tax_incl = (float) $orderDetail->total_price_tax_incl + $totalAmount;
+                    }
+
+                    Db::getInstance()->insert('order_detail_tax', $orderDetailTaxes, false);
+                }
+
+                // Update OrderDetail values
+                $orderDetail->unit_price_tax_incl = (float) $orderDetail->unit_price_tax_excl + $unitAmount;
+                $orderDetail->total_price_tax_incl = (float) $orderDetail->total_price_tax_incl + $totalAmount;
+                if (!$orderDetail->update()) {
+                    throw new OrderException('An error occurred while editing the product line.');
+                }
+            }
+        } finally {
+            $this->contextStateManager->restorePreviousContext();
+        }
+    }
+
+    /**
+     * @param Order $order
+     *
+     * @return array
+     */
+    private function prepareOrderContext(Order $order): array
+    {
+        $shopConstraint = new ShopConstraint(
+            (int) $order->id_shop,
+            (int) $order->id_shop_group
+        );
+        $roundType = (int) $this->shopConfiguration->get('PS_ROUND_TYPE', null, $shopConstraint);
```

Comment:
```suggestion
        $roundType = $this->shopConfiguration->getInt('PS_ROUND_TYPE', null, $shopConstraint);
```

Reply:
Nope because I use the interface not the adapter class